w3af/core/data/parsers/doc/open_api/specification.py

```diff
@@ -108,13 +109,23 @@ def _set_operation_params(self, operation):
     def _parse_spec_from_dict(self, spec_dict, retry=True):
         """
         load_spec_dict will load the open api document into a dict. We use this
-        function to parse the dict into a bravado Spec instance.
+        function to parse the dict into a bravado Spec instance. By default,
+        it validates the spec, but validation may be disabled
+        by passing `no_validation=True` to the construction
 
         :param spec_dict: The output of load_spec_dict
         :return: A Spec instance which holds all the dict information in an
                  accessible way.
         """
         config = {'use_models': False}
+        if self.no_validation:
+            om.out.debug('Open API spec validation disabled')
+            config.update({
```

Comment:
If the validation fails, then basically `Spec.from_dict()` returns `None`, and as a result, no API endpoints are added for further testing. If I understand correctly, any exceptions below are logged only if debug mode is enabled. Here is an example of a error I saw (it may depend on issues in a particular API spec):

`The document at "file:///path/to/swagger.yaml" is not a valid Open API specification. The following exception was raised while parsing the dict into a specification object: "('expected string or buffer', TypeError('expected string or buffer',))"
`

Although I didn't see any additional info. I am not sure if printing out a full stacktrace would help, and I also didn't find a way how to enable additional logs in bravado. Maybe we can always print a warning if the validation fails, not only when debug mode is enabled. What do you think?

Reply:
With that log line we're fine. Thanks for the detailed answer.